cmd/ci-operator/main.go

```diff
@@ -1385,18 +1386,41 @@ func (o *options) saveNamespaceArtifacts() {
 	}
 }
 
+func loadLeaseCredentials(leaseServerUsername, leaseServerPasswordFile, leaseServerCredentialsFile string) (string, func() []byte, error) {
+	sa := &secret.Agent{}
+	if leaseServerCredentialsFile != "" {
+		if err := sa.Start([]string{leaseServerCredentialsFile}); err != nil {
+			return "", nil, fmt.Errorf("failed to start secret agent on file %s: %s", leaseServerCredentialsFile, string(sa.Censor([]byte(err.Error()))))
+		}
+		splits := strings.Split(string(sa.GetSecret(leaseServerCredentialsFile)), ":")
+		if len(splits) != 2 {
+			return "", nil, fmt.Errorf("got invalid content of lease server credentials file which must be of the form '<username>:<passwrod>'")
+		}
+		username := splits[0]
+		passwordGetter := func() []byte {
+			splits := strings.Split(string(sa.GetSecret(leaseServerCredentialsFile)), ":")
+			if len(splits) != 2 {
+				log.Printf("got invalid content of lease server credentials file which must be of the form '<username>:<passwrod>'")
+				return nil
+			}
+			return []byte(splits[1])
+		}
+		return username, passwordGetter, nil
```

Comment:
Hm what is fishy here is that the password getter is IMO unnecessarily dynamic - if the underlying file can change (which it IMO not expected for ci-operator), then the username can change too. So I'd just read the file once here and made the password getter a simple 

```go
func() []byte {
  return []byte(splits[1])
}
```

and not pretend to allow credentials reloading when it really doesn't.

Reply:
Updated.